pkg/common/providers/ocmprovider/cluster.go

```diff
@@ -19,6 +19,46 @@ import (
 	"github.com/spf13/viper"
 )
 
+// IsValidClusterName validates the clustername prior to proceeding with it
+// in launching a cluster.
+func (o *OCMProvider) IsValidClusterName(clusterName string) (bool, error) {
+	// Create a context:
+	ctx := context.Background()
+
+	collection := o.conn.ClustersMgmt().V1().Clusters()
+
+	// Retrieve the list of clusters using pages of ten items, till we get a page that has less
+	// items than requests, as that marks the end of the collection:
+	size := 10
+	page := 1
+	searchPhrase := fmt.Sprintf("name = '%s'", clusterName)
```

Comment:
Why is this different from the MOA query above? Is there a difference between the two?

This query would lead me to think it should only ever return a single result. As far as I know, when someone logs in to OCM they're part of an Org, and no Org can have a cluster name collision (CS won't allow it.)

So theoretically you should be able to remove the for loop and then validate if there are zero or not-zero results.

Reply:
> Why is this different from the MOA query above? Is there a difference between the two?
Please see previous comment

The loop accounts for the size and page requests. I used the size/page ints from the examples in the OCM sdk. We could probabaly enlarge the size to avoid more loops. However the for loop should stay IMO as when production fleet is at a large number, we will need to iterate through the pages. e819298f